src/mongoose_hooks.erl

```diff
@@ -500,14 +500,15 @@ xmpp_send_element(HostType, Acc, El) ->
 
 %%% @doc The `xmpp_stanza_dropped' hook is called to inform that
 %%% an xmpp stanza has been dropped.
--spec xmpp_stanza_dropped(HostType, From, To, Packet) -> Result when
+-spec xmpp_stanza_dropped(HostType, From, To, Packet, Acc) -> Result when
     HostType :: binary(),
     From :: jid:jid(),
     To :: jid:jid(),
     Packet :: exml:element(),
+    Acc :: mongoose_acc:t(),
     Result :: any().
-xmpp_stanza_dropped(HostType, From, To, Packet) ->
-    ejabberd_hooks:run_for_host_type(xmpp_stanza_dropped, HostType, ok,
+xmpp_stanza_dropped(HostType, From, To, Packet, Acc) ->
+    ejabberd_hooks:run_for_host_type(xmpp_stanza_dropped, HostType, Acc,
```

Comment:
How about taking the host type from `Acc`? Many other hooks do that already and the metrics handler relies on the host type anyway:
```suggestion
xmpp_stanza_dropped(Acc, From, To, Packet) ->
    HostType = mongoose_acc:host_type(Acc),
    ejabberd_hooks:run_for_host_type(xmpp_stanza_dropped, HostType, Acc,
```

Reply:
I was also think about it, but need to check if this is suite for `mod_amp:take_action_for_matched_rule/4` - by some reason the `Host` is get from `From#jid.lserver`

Reply:
Done. Thanks.